### ISSUE TITLE
Windows 7: `update.exe` fails to start target app on `apply` and `start`

### ISSUE THREAD

**Issue by Trinitek** [velopack/velopack#138]
Version 0.0.530-g4f25004
win7-x64, non-ESU

This also affects 0.0.474-gb9d73fc which is my previously last-known version to work on Win7.

The log file contains results for consecutive runs of `update.exe apply` and `update.exe start` from the app directory. The commands behave as expected on win10.0.14393-x86 and win10.0.22631-x64.

```log
[update:1172] [06:37:00] [INFO] --
[update:1172] [06:37:00] [INFO] Starting Velopack Updater (0.0.530-g4f25004)
[update:1172] [06:37:00] [INFO]     Location: C:\Users\trinitek\AppData\Local\MyApp\Update.exe
[update:1172] [06:37:00] [INFO]     CWD: C:\Users\trinitek\AppData\Local\MyApp
[update:1172] [06:37:00] [INFO]     Verbose: false
[update:1172] [06:37:00] [INFO]     Silent: false
[update:1172] [06:37:00] [INFO]     Log File: None
[update:1172] [06:37:00] [INFO] Command: Apply
[update:1172] [06:37:00] [INFO]     Restart: true
[update:1172] [06:37:00] [INFO]     Wait: NoWait
[update:1172] [06:37:00] [INFO]     Package: None
[update:1172] [06:37:00] [INFO]     Exe Args: None
[update:1172] [06:37:00] [INFO] Loaded manifest for application: MyApp
[update:1172] [06:37:00] [INFO] Root Directory: C:\Users\trinitek\AppData\Local\MyApp
[update:1172] [06:37:00] [INFO] Attempting to open global system mutex: 'velopack-MyApp'
[update:1172] [06:37:00] [INFO] Attempting to auto-detect package in: C:\Users\trinitek\AppData\Local\MyApp\packages
[update:1172] [06:37:00] [INFO] Found 1.8925.1: 'C:\Users\trinitek\AppData\Local\MyApp\packages\MyApp-1.8925.1-full.nupkg'
[update:1172] [06:37:00] [INFO] Getting ready to apply package to MyApp ver 1.8925.1: C:\Users\trinitek\AppData\Local\MyApp\packages\MyApp-1.8925.1-full.nupkg
[update:1172] [06:37:00] [INFO] Applying package to current: 1.8925.1
[update:1172] [06:37:00] [INFO] Checking application pre-requisites...
[update:1172] [06:37:00] [INFO]     .NET Framework 4.6.2 is already installed.
[update:1172] [06:37:00] [INFO] Running --veloapp-obsolete hook...
[update:1172] [06:37:00] [WARN] Error running hook --veloapp-obsolete: The handle is invalid. (os error 6) (took 0.0592ms)
[update:1172] [06:37:00] [INFO] Checking for running processes in: C:\Users\trinitek\AppData\Local\MyApp
[update:1172] [06:37:00] [WARN] Skipping killing self: C:\Users\trinitek\AppData\Local\MyApp\Update.exe (1172)
[update:1172] [06:37:00] [INFO] Extracting bundle to C:\Users\trinitek\AppData\Local\MyApp\packages\tmp_YsQ8cLng
[update:1172] [06:37:00] [INFO] Extracting 238 app files...
[update:1172] [06:37:02] [INFO] Checking for running processes in: C:\Users\trinitek\AppData\Local\MyApp
[update:1172] [06:37:02] [WARN] Skipping killing self: C:\Users\trinitek\AppData\Local\MyApp\Update.exe (1172)
[update:1172] [06:37:02] [INFO] Replacing bundle at C:\Users\trinitek\AppData\Local\MyApp\current
[update:1172] [06:37:02] [INFO] Bundle extracted successfully to C:\Users\trinitek\AppData\Local\MyApp\current
[update:1172] [06:37:02] [INFO] Writing uninstall registry key...
[update:1172] [06:37:02] [INFO] Running --veloapp-updated hook...
[update:1172] [06:37:02] [WARN] Error running hook --veloapp-updated: The handle is invalid. (os error 6) (took 0.0556ms)
[update:1172] [06:37:02] [INFO] Checking for running processes in: C:\Users\trinitek\AppData\Local\MyApp
[update:1172] [06:37:02] [WARN] Skipping killing self: C:\Users\trinitek\AppData\Local\MyApp\Update.exe (1172)
[update:1172] [06:37:02] [INFO] Package applied successfully.
[update:1172] [06:37:02] [INFO] Package version 1.8925.1 applied successfully.
[update:1172] [06:37:02] [INFO] About to launch: 'C:\Users\trinitek\AppData\Local\MyApp\current\MyApp.exe' in dir 'C:\Users\trinitek\AppData\Local\MyApp\current'
[update:1172] [06:37:02] [INFO] Args: CommandArgs { inner: [] }
[update:1172] [06:37:02] [ERROR] Apply error: Failed to start application (The handle is invalid. (os error 6)).
[update:2956] [06:37:06] [INFO] --
[update:2956] [06:37:06] [INFO] Starting Velopack Updater (0.0.530-g4f25004)
[update:2956] [06:37:06] [INFO]     Location: C:\Users\trinitek\AppData\Local\MyApp\Update.exe
[update:2956] [06:37:06] [INFO]     CWD: C:\Users\trinitek\AppData\Local\MyApp
[update:2956] [06:37:06] [INFO]     Verbose: false
[update:2956] [06:37:06] [INFO]     Silent: false
[update:2956] [06:37:06] [INFO]     Log File: None
[update:2956] [06:37:06] [INFO] Command: Start
[update:2956] [06:37:06] [INFO]     Wait: NoWait
[update:2956] [06:37:06] [INFO]     Exe Name: None
[update:2956] [06:37:06] [INFO]     Exe Args: None
[update:2956] [06:37:06] [INFO] Loaded manifest for application: MyApp
[update:2956] [06:37:06] [INFO] Root Directory: C:\Users\trinitek\AppData\Local\MyApp
[update:2956] [06:37:06] [INFO] Attempting to open global system mutex: 'velopack-MyApp'
[update:2956] [06:37:06] [INFO] Loaded manifest for application: MyApp
[update:2956] [06:37:06] [INFO] Root Directory: C:\Users\trinitek\AppData\Local\MyApp
[update:2956] [06:37:06] [INFO] About to launch: 'C:\Users\trinitek\AppData\Local\MyApp\current\MyApp.exe' in dir 'C:\Users\trinitek\AppData\Local\MyApp\current'
[update:2956] [06:37:06] [ERROR] Start error: The handle is invalid. (os error 6)
```

**Comment by Trinitek** [comment]
I can reproduce this error code with `Setup.exe`. The setup program is supposed to execute post-install hooks and then launch the app. The post-setup launch succeeds when I double-click the application from a File Explorer window. However, I can cause `os error 6` for the post-setup launch if I start it from a command prompt.

**Comment by caesay** [comment]
Fixed in 0.0.541-gf8f3de0 (f8f3de0b6512423efc128fa697165603f93f4578)

This is because on Win7, a x64 process can not inherit handles (eg. stdout/stderr) from an x86 process. Velopack is x86 for compatibility reasons.

**Comment by Trinitek** [comment]
👍 Fix confirmed on win7-x64 non-ESU.
